### ISSUE TITLE
Dashboard showing incorrect Scheduled Transaction

### ISSUE THREAD

**Issue by spillaip** [moneymanagerex/moneymanagerex#6680]
MMEX version:
 - [ ] 1.6.4
 - [] 1.7.0
 - [x ] 1.7.1
 - [ ] 1.7.2
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug
Dashboard is indicating a scheduled transaction as due, while the next instance is expected in 2025.

Please describe your issue with details.
Add screenshot or other files if needed.
[Incorrect Schedule.pdf](https://github.com/moneymanagerex/moneymanagerex/files/15110601/Incorrect.Schedule.pdf)

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

Expected result:
This scheduled transaction should not be displayed
Actual result:
Able to see the transaction with past due date
# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by rmelillo76** [comment]
Confirmed, I can reproduce this. It appears the "Date Paid" rather than the "Date Due" is driving the Upcoming Transaction list. 
I'm assuming your date format is YYYY-DD-MM, so 06-01-2024 (Jan 1st) was 110 days ago. 
I have many schedule transactions and all have the same date in both, so I've never noticed this before.

**Comment by n-stein** [comment]
I think this is intentional. You'll notice that if "Date Paid" is in the past it will say "Delayed" and if "Date Due" is in the past it will say "Overdue". These mean different things.

"Delayed" means the transaction date is in the past -- you are delaying entering a transaction which has already occurred according to the TRANSDATE.

"Overdue" means the scheduled ("Due") date is in the past -- the transaction should have been entered according to your schedule, even if the Transaction Date is technically in the future.

**Comment by rmelillo76** [comment]
Ahhh @n-stein, great observation!

**Comment by spillaip** [comment]
@n-stein The bill was paid already on the same day.  I expect there is no entry in the schedule region of the dashboard for already paid scheduled entries.  This is confusing.

**Comment by rmelillo76** [comment]
> @n-stein The bill was paid already on the same day.  I expect there is no entry in the schedule region of the dashboard for already paid scheduled entries.  This is confusing.

@spillaip did you use the schedule transaction screen to enter the transaction or did you just input it manually? Since the transaction that is "scheduled" is already done you can use the skip option or modify the date to reflect the next day it is expected to be paid.

**Comment by spillaip** [comment]
@rmelillo76 @n-stein The transaction was processed as import, the expectation is that the scheduled transactions are marked done during this process based on certain criteria.

**Comment by rmelillo76** [comment]
It's an interesting idea, but the system does not have any logic to do that. There are many ways to add a new transaction, importing is just one of them - manual input, copy & paste, duplicate, etc.  None of those methods correlate the entry back to a scheduled transaction, they all are treated independent of each other. 

During an import, I think it would be even more complicated since the file would normally contain multiple transactions that would have to be scanned against the scheduled list looking for an exact match on all fields? Many files that are downloaded from your bank will contain variable transaction descriptions, and that could be difficult to match as well from a programming standpoint.

**Comment by spillaip** [comment]
@rmelillo76 I appreciate your effort to analyze the solution.  I have an idea that finally the transaction hits the ledger irrespective of the method.  So, just prior to the ledger, compare the transaction against the scheduled transaction.

Alternately, in the import dialog, introduce a checkpoint to validate against the scheduled transactions (similar to matching payee and notes".

Please note that I am not a programming expert, I may not know the intricacies involved in implementing them.
